add a SL faq addressing data platforms (#4968)
fishtown-analytics/docs.getdbt.com: website/docs/docs/use-dbt-semantic-layer/sl-faqs.md

## Availability

<expandable alt_header="What data platforms are supported by the dbt Semantic Layer?">

> The dbt Semantic Layer supports the following data platforms: Snowflake, BigQuery, Databricks, or Redshift.

Reviewer: I think we would normally do something like this as a bulleted list, yeah?
```suggestion
The dbt Semantic Layer supports the following data platforms: 

- Snowflake 
- BigQuery 
- Databricks  
- Redshift
```

Author: yea its not linking or steps. no strong feelings so pushing it forward and merging the pr 🙌

> Support for other data platforms, such as Fabric and Trino, isn't available at this time. If you're interested in using the dbt Semantic Layer with a different data platform, please reach out to us [here](https://www.getdbt.com/get-started).

Reviewer: Maybe we don't need to call out platforms we *don't* support.
I don't think that link is going to be helpful as a contact link. Is there something more direct we could use like and email address or a link to a page with contact info?

```suggestion
Support for other data platforms, such as Fabric and Trino, isn't available at this time. If you're interested in using the dbt Semantic Layer with a data platform not on the list, please [contact us](https://www.getdbt.com/get-started).
```

Author: i think it's helpful to be explicit and not encourage inferring, personally imho. esp since users/sa's asked about it. 

i like the mesh of:
> Support for other data platforms, such as Fabric and Trino, isn't available at this time. If you're interested in using the dbt Semantic Layer with a data platform not on the list, please [contact us](https://www.getdbt.com/get-started).
